### ISSUE TITLE
Variance reduction for (Renyi)ELBO

### ISSUE THREAD

**Issue by OlaRonning** [pyro-ppl/pyro#3104]
From the forum [thread](https://forum.pyro.ai/t/iwae-elbo-in-pyro-numpyro/3604/2). [Tighter Variational Bounds are Not Necessarily Better](https://arxiv.org/pdf/1802.04537.pdf) demonstrates that tighter VBs can lead to deteriorating gradient estimates for VAEs trained with [IWAE](https://arxiv.org/pdf/1509.00519.pdf) due to the variance decreasing slower than the expected value. [DReG](https://arxiv.org/pdf/1810.04152.pdf) is a variance reduction technique that can alleviate this issue. 

@fritzo and @eb8680 have a DReG draft for RenyiELBO (#2605), from feature request #2598. Is it a good place to continue from?

@martinjankowiak you mentioned there may be other ways to revamp the ELBOs, did you have something particular in mind?

**Comment by martinjankowiak** [comment]
@OlaRonning well i believe the current renyi implementation is suboptimal (or incorrect, depending on your point of view) for a local latent variable model like a VAE. basically it has to do with order of summation/logarithm. if there are plates that enclose the entire model then the corresponding summations should be moved out but as it is i believe the implementation effectively assumes that there is a single global latent variable and doesn't make effective use of conditional independence

not sure if you should start from [2605](https://github.com/pyro-ppl/pyro/pull/2605 ) but you should definitely look at it carefully :)

**Comment by davidaknowles** [comment]
@martinjankowiak just to clarify is the RenyiELBO implementation just "incorrect" in terms of being higher variance (for a fixed `num_particles`) than it should be? i.e. it will still correctly estimate the true marginal likelihood for sufficiently large `num_particles`?

**Comment by martinjankowiak** [comment]
@davidaknowles  yes, that's right. but in practice the variance is expected to be very high and `num_particles` would need to be very very large to compensate. as you can see there is an open PR but it was never merged...

**Comment by davidaknowles** [comment]
Thanks that's helpful to know. I mostly just want it for getting test marginal likelihood where I can just throw a few thousand `num_particles` at it, rather than for training.

**Comment by martinjankowiak** [comment]
yes though in that case i'd probably do it one data point at a time if that is feasible
